garage/tf/q_functions/continuous_mlp_q_function.py

```diff
@@ -59,19 +57,22 @@ def __init__(self,
         self._action_merge_layer = action_merge_layer
         self._output_nonlinearity = output_nonlinearity
         self._batch_norm = bn
+        self._reuse = reuse
+        self._trainable = trainable
+        self._f_qval, self._output_layer, self._obs_layer, self._action_layer = self._build_net(
+            reuse=self._reuse, trainable=self._trainable, name=self.name)
+        LayersPowered.__init__(self, [self._output_layer])
 
-    def _build_net(self, reuse=None, custom_getter=None, trainable=None):
+    def _build_net(self, reuse=None, trainable=None, name=None):
         """
         Set up q network based on class attributes. This function uses layers
-        defined in rllab.tf.
+        defined in garage.tf.
 
         Args:
             reuse: A bool indicates whether reuse variables in the same scope.
-            custom_getter: A customized getter object used to get variables.
             trainable: A bool indicates whether variables are trainable.
         """
-        with tf.variable_scope(
-                self.name, reuse=reuse, custom_getter=custom_getter):
+        with tf.variable_scope(name, reuse=reuse):
```

Comment:
I think you should create this scope once in the constructor and then reuse it whenever you need it.

Reply:
See the style used in my version of GaussianMLPPolicy.

Reply:
I removed `reuse`. I created the scope in the constructor. I need to move this in `_build_net()` because it is used to create the target network. I call `self.policy._build_net(trainable=False, name="target_policy")` to build the target network graph.